client/components/domains/domain-suggestion/index.jsx

```diff
@@ -1,57 +1,53 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React from 'react';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-var DomainProductPrice = require( 'components/domains/domain-product-price' ),
-	abtest = require( 'lib/abtest' ).abtest,
-	Gridicon = require( 'components/gridicon' );
+import DomainProductPrice from 'components/domains/domain-product-price';
 
-var DomainSuggestion = React.createClass( {
+const DomainSuggestion = React.createClass( {
 
 	propTypes: {
-		buttonLabel: React.PropTypes.string,
+		buttonContent: React.PropTypes.oneOfType( [ React.PropTypes.string, React.PropTypes.element ] ).isRequired,
 		buttonClasses: React.PropTypes.string,
 		extraClasses: React.PropTypes.string,
-		onButtonClick: React.PropTypes.func,
+		onButtonClick: React.PropTypes.func.isRequired,
+		priceRule: React.PropTypes.string.isRequired,
 		price: React.PropTypes.string,
-		cart: React.PropTypes.object,
-		isAdded: React.PropTypes.bool.isRequired,
-		withPlansOnly: React.PropTypes.bool
+		cart: React.PropTypes.object.isRequired,
+		domain: React.PropTypes.string
 	},
 
-	renderButton: function() {
-		var buttonContent;
-		if ( this.props.isAdded ) {
-			buttonContent = <Gridicon icon="checkmark" ref="checkmark" />;
-		} else {
-			buttonContent = abtest( 'domainsWithPlansOnly' ) === 'plansOnly' && ! this.props.price ? this.translate( 'Select' ) : this.props.buttonLabel;
-		}
+	renderButton() {
```

Comment:
Nice simplification of this method! You can really see the benefits of your approach here :)